packages/calcite-components/src/utils/sortableComponent.ts

```diff
@@ -12,27 +18,100 @@ export interface SortableComponent {
    */
   readonly el: HTMLElement;
 
+  /**
+   * When `true`, dragging is enabled.
+   */
+  dragEnabled: boolean;
+
+  /**
+   * Specifies which items inside the element should be draggable.
+   */
+  dragSelector?: string;
+
+  /**
+   * The list's group identifier.
+   */
+  group?: string;
+
+  /**
+   * The selector for the handle elements.
+   */
+  handleSelector: string;
+
   /**
    * The Sortable instance.
    */
   sortable: Sortable;
+
+  /**
+   * Whether the element can  move from the list.
+   */
+  dragCanPull: (event: DragEvent) => boolean;
+
+  /**
+   * Whether the element can be added from another list.
+   */
+  dragCanPut: (event: DragEvent) => boolean;
+
+  /**
+   * Called by any change to the list (add / update / remove).
+   */
+  onDragSort: (event: Sortable.SortableEvent) => void;
+
+  /**
+   * Element dragging started.
+   */
+  onDragStart?: (event: Sortable.SortableEvent) => void;
+
+  /**
+   * Element dragging ended.
+   */
+  onDragEnd?: (event: Sortable.SortableEvent) => void;
 }
 
 /**
  * Helper to keep track of a SortableComponent. This should be called in the `connectedCallback` lifecycle method as well as any other method necessary to rebuild the sortable instance.
  *
  * @param {SortableComponent} component - The sortable component.
- * @param {SortableComponent} [options] - Sortable options object.
  */
-export function connectSortableComponent(component: SortableComponent, options?: Sortable.Options): void {
+export function connectSortableComponent(component: SortableComponent): void {
   disconnectSortableComponent(component);
   sortableComponentSet.add(component);
 
   if (inactiveSortableComponentSet.has(component)) {
     return;
   }
 
-  component.sortable = Sortable.create(component.el, options);
+  const dataIdAttr = "id";
+  const { group, handleSelector: handle, dragSelector: draggable } = component;
+
+  component.sortable = Sortable.create(component.el, {
```

Comment:
Can we type the callback functions in this init object?

Reply:
These are typed by Sortable already. 🎉